respy/shared.py

```diff
@@ -264,3 +263,30 @@ def normalize_probabilities(probabilities):
     probabilities[-1] = 1 - probabilities[:-1].sum()
 
     return probabilities
+
+
+@nb.guvectorize(
+    ["f8, f8, f8, f8, f8, b1, f8[:], f8[:]"],
+    "(), (), (), (), (), () -> (), ()",
+    nopython=True,
+    target="parallel",
+)
+def calculate_value_functions_and_flow_utilities(
+    wage,
+    nonpec,
+    continuation_value,
+    draw,
+    delta,
+    is_inadmissible,
+    value_function,
+    flow_utility,
+):
+    """Calculate the choice-specific value functions and flow utilities.
+
```

Comment:
Why not copy the more detailed docstring of the old function?

Reply:
I extended the docstring. The reason is that this function is just a wrapper around ``aggregate_keane_wolpin_utility`` for arrays with arbitrary dimensions and I wanted to avoid repitition. I made it clear that the functions belong together.